pkg/datapath/loader/config/config.go

```diff
@@ -0,0 +1,16 @@
+package config
+
+import (
+	"github.com/cilium/cilium/pkg/datapath"
+	"github.com/sirupsen/logrus"
+)
+
+// EndpointConfig provides access to EndpointConfig information that is necessary to
+// compile and load the datapath.
+type EndpointConfig interface {
```

Comment:
What's different conceptually between this `EndpointConfig` and  the nested `datapath.EndpointConfiguration`? They have very similar names.

Can (should?) we just extend the datapath.EndpointConfiguration type to have these extra bits since this new package is introducing the dependency on the datapath package anyway?

Reply:
I factored out the `endpoint` interface out of `pkg/loader` into its own `Endpoint` interface within `pkg/datapath`. I'll look into seeing if I can collapse the two interfaces together, that's a good idea.